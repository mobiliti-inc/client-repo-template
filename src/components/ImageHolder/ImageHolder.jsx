import React, { Fragment, PureComponent } from 'react';
import classNames from 'classnames';

import styles from './ImageHolder.scss';
import { iconEdit } from '../../assets/icons/';

class ImageHolder extends PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			loaded: false,
		};
		this.image = React.createRef();
	}

	componentDidMount() {
		const img = this.image.current;
		if (img && img.complete) {
			this.handleImageLoaded();
		}
	}

	handleImageLoaded = () => {
		if (!this.state.loaded) {
			this.setState({ loaded: true });
		}
	}

	renderImageInput = () => (
		<input
			styleName="input-upload-image"
			type="file"
			id={this.props.id || 'upload-image'}
			accept="image/jpeg"
			onChange={this.props.onUploadImageInputChange}
		/>
	);

	render() {
		const {
			onUploadImageClick,
			imagefile,
			squared,
			placeholder,
			imagePlaceholder,
		} = this.props;
		if (!imagefile && placeholder) {
			return (
				<Fragment>
					{this.renderImageInput()}
					{placeholder}
				</Fragment>
			);
		}
		return (
			<div styleName={classNames('image-container', { 'image-container--squared': squared })}>
				{imagefile && <img src={imagefile} ref={this.image} alt="Holder" onLoad={this.handleImageLoaded} styleName="image" />}
				{!this.state.loaded && imagePlaceholder && <img src={imagePlaceholder} alt="Placeholder" styleName="image-placeholder" />}
				<span
					role="presentation"
					styleName="edit-icon"
					onClick={onUploadImageClick}
				>
					<img src={iconEdit} alt="edit icon" />
					{this.renderImageInput(...this.props)}
				</span>
			</div>
		);
	}
}

ImageHolder.propTypes = {
	onUploadImageClick: PropTypes.func.isRequired,
	onUploadImageInputChange: PropTypes.func.isRequired,
	imagefile: PropTypes.string,
	placeholder: PropTypes.node,
	squared: PropTypes.bool,
	id: PropTypes.string,
	imagePlaceholder: PropTypes.string,
};

ImageHolder.defaultProps = {
	squared: false,
	placeholder: null,
	imagefile: null,
	id: '',
	imagePlaceholder: '',
};

export default CSSModules(ImageHolder, styles, { allowMultiple: true });
