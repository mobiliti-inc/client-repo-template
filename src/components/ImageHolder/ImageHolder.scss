@import "~variables";

.image-container {
    display: flex;
    height: 191px;
    width: 191px;
    position: relative;
    border-radius: 100%;
    justify-content: center;

    .image {
        height: 100%;
        width: 100%;
        border-radius: 100%;
    }

    &--squared {
        border-radius: 5px;

        .image {
            border-radius: 5px;
            height: 154px;
        }

        .image-placeholder {
            width: 175px;
            height: 120px;
        }

        .edit-icon {
            top: -25px;
        }
    }
}

.edit-icon {
    display: flex;
    position: absolute;
    height: 48px;
    width: 48px;
    top: 15px;
    right: -8px;
    background: #FFFFFF;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 100%;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    &>img {
        width: 19.03px;
        height: 20.04px;
        object-fit: contain;
    }
}

.input-upload-image {
    display: none;
}