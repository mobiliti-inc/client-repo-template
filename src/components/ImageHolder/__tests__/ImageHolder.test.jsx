import ImageHolder from '../ImageHolder';


describe('ImageHolder', () => {
	const defaultProps = {
		onUploadImageClick: jest.fn(),
		onUploadImageInputChange: jest.fn(),
		imagefile: ''
	};
	it('renders', () => {
		const tree = shallow(<ImageHolder {...defaultProps} />);
		matchSnapshot(tree);
	});
});
