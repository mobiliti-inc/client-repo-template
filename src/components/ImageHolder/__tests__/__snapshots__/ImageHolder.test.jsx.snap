// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageHolder renders 1`] = `
<div
  className="image-container"
>
  <span
    className="edit-icon"
    onClick={[MockFunction]}
    role="presentation"
  >
    <img
      alt="edit icon"
      src="test-file-stub"
    />
    <input
      accept="image/jpeg"
      className="input-upload-image"
      id="upload-image"
      onChange={[MockFunction]}
      type="file"
    />
  </span>
</div>
`;
